File: python/키패드 누르기.py
def solution(numbers, hand):
    answer = ''

    keyPad = [[1, 2, 3], [4, 5, 6], [7, 8, 9], ['*', 0, '#']]

    leftHand = {'x': 3, 'y': 0}
    rightHand = {'x': 3, 'y': 2}

    for n in numbers:
        if n == 2 or n == 5 or n == 8 or n == 0:
            x = 0
            y = 1
            if n == 2:
                x = 0
                y = 1
            if n == 5:
                x = 1
                y = 1
            if n == 8:
                x = 2
                y = 1
            if n == 0:
                x = 3
                y = 1
            leftDist = abs(x-leftHand['x'])+abs(y-leftHand['y'])
            rightDist = abs(x-rightHand['x'])+abs(y-rightHand['y'])
            if leftDist == rightDist:
                if hand == "right":
                    answer += "R"
                    rightHand['x'] = x
                    rightHand['y'] = y

                else:
                    answer += "L"
                    leftHand['x'] = x
                    leftHand['y'] = y

            else:
                if leftDist < rightDist:
                    answer += "L"
                    leftHand['x'] = x
                    leftHand['y'] = y

                elif leftDist > rightDist:
                    answer += "R"
                    rightHand['x'] = x
                    rightHand['y'] = y

        elif n == 1 or n == 4 or n == 7:
            answer += "L"
            if n == 1:
                leftHand['x'] = 0
                leftHand['y'] = 0
            if n == 4:
                leftHand['x'] = 1
                leftHand['y'] = 0
            if n == 7:
                leftHand['x'] = 2
                leftHand['y'] = 0
        else:
            if n == 3:
                rightHand['x'] = 0
                rightHand['y'] = 2

            if n == 6:
                rightHand['x'] = 1
                rightHand['y'] = 2
            if n == 9:
                rightHand['x'] = 2
                rightHand['y'] = 2
            answer += "R"

    return answer

File:  cpp/키패드 누르기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;
struct right {
    int x, y;
};

struct left {
    int x, y;
};

struct keypad {
    int x, y;
};
string solution(vector<int> numbers, string hand) {
    string answer = "";
    int n=1;
    keypad keypad[10];
    for(int i=0; i<3; i++){
        for(int j=0; j<3; j++){
            keypad[n].x=i;
            keypad[n].y=j;
            n++;
        }
    }
    keypad[0].x=3;
    keypad[0].y=1;
    
    right r;
    left l;
    
    l={3, 0};
    r={3, 2};
    
    
    for(int i=0; i<numbers.size(); i++){
        if(numbers[i]==1 || numbers[i]==4 || numbers[i]==7){
            int val=numbers[i];
            l={keypad[val].x, keypad[val].y};
            answer+="L";
        }else if(numbers[i]==3 || numbers[i]==6 || numbers[i]==9){
            int val=numbers[i];
            r={keypad[val].x, keypad[val].y};
            answer+="R";
        }else{
            int val=numbers[i];
            int ld=abs(l.x-keypad[val].x)+abs(l.y-keypad[val].y);
            int rd=abs(r.x-keypad[val].x)+abs(r.y-keypad[val].y);
            if(ld>rd){
                answer+="R";
                r={keypad[val].x, keypad[val].y};
            }else if(ld<rd){
                answer+="L";
                l={keypad[val].x, keypad[val].y};
            }else{
                if(hand=="right"){
                    answer+="R";
                    r={keypad[val].x, keypad[val].y};
                }else{
                    answer+="L";
                    l={keypad[val].x, keypad[val].y};
                }
            }
        }
    }
    
    
    
    
    return answer;
}